common/metrics/otel_metrics_handler.go

```diff
@@ -35,8 +35,6 @@ import (
 	"go.temporal.io/server/common/log/tag"
 )
 
-// MetricsHandler is an event.Handler for OpenTelemetry metrics.
-// Its Event method handles Metric events and ignores all others.
```

Comment:
maybe write a new comment explaining what otelMetricsHandler is? it's pretty obvious, but still might be nice